test/index.js

```diff
@@ -1902,4 +1902,37 @@ describe('normalizr', function () {
     }).should.throw();
   });
 
-});
+  it('can normalize iterables keyed with their id', function () {
+    var user = new Schema('users', {
+      idAttribute: function(obj, key) {
+        return key;
+      }
+    })
+
+    var input = {
+      1: {
+        name: 'Adam'
+      },
+      4: {
+        name: 'Jeremy'
+      }
+    };
+
+    Object.freeze(input);
+
+    normalize(input, arrayOf(user)).should.eql({
```

Comment:
`arrayOf` seems like a poor use for an object. Would it simplify things to add a `keyedObject` function that assumes your `ID` is the `key` for each entry?